vertical-pod-autoscaler/pkg/recommender/input/controller_fetcher/controller_fetcher.go

```diff
@@ -82,10 +83,41 @@ type controllerFetcher struct {
 	scaleNamespacer scale.ScalesGetter
 	mapper          apimeta.RESTMapper
 	informersMap    map[wellKnownController]cache.SharedIndexInformer
+	controllerCache controllerCacheStorage
+}
+
+func (f *controllerFetcher) periodicallyRemoveExpired(ctx context.Context, period time.Duration) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(period):
+			f.controllerCache.RemoveExpired()
+		}
+	}
+}
+
+func (f *controllerFetcher) periodicallyRefresh(ctx context.Context, period time.Duration) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(period):
+			for _, item := range f.controllerCache.GetKeysToRefresh() {
```

Comment:
Could you please add some logs here marking start & finish of this procedure? Ideally with the number of entries refreshed after the loop.

Reply:
Done.